components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementPolicyAdminService.java

```diff
@@ -285,11 +278,11 @@ public PolicyDTO getPolicy(String policyId, boolean isPDPPolicy) throws Entitlem
      */
     public PolicyDTO getPolicyByVersion(String policyId, String version) throws EntitlementException {
 
-        PolicyDTO policyDTO = null;
+        PolicyDTO policyDTO;
 
         try {
-            PolicyVersionManager versionManager = EntitlementAdminEngine.getInstance().getVersionManager();
-            policyDTO = versionManager.getPolicy(policyId, version);
+            PolicyDAO policyStore = new RegistryPolicyDAOImpl();
```

Comment:
Should handle similar to https://github.com/wso2/carbon-identity-framework/pull/5673/files#r1594986499